src/browser/atom-window.coffee

```diff
@@ -141,14 +141,15 @@ class AtomWindow
     @browserWindow.webContents.on 'crashed', =>
       global.atomApplication.exit(100) if @headless
 
-      chosen = dialog.showMessageBox @browserWindow,
+      dialog.showMessageBox @browserWindow,
```

Comment:
As described in de599f9c66b1838540cfd263754225908a237048, we need to make this dialog asynchronous so that other `crashed` event handlers (i.e. the recovery service ones) have the chance to run.